hvantk/commands/ptm_cli.py

```
@@ -691,3 +691,337 @@ def ptm_constraint(
+    if sep is None:
+        # Fall back to comma separator if neither .tsv nor .csv explicit.
+        sep = "," if p.endswith((".csv", ".csv.gz")) else "\t"
+    return _pd.read_csv(p, sep=sep, low_memory=False)
+
+
+def _read_expression_wide(pkl_path, tsv_path):
+    """Load a gene x stratum expression matrix from a pandas pickle or TSV."""
+    import pandas as _pd
+
+    if pkl_path:
+        wide = _pd.read_pickle(pkl_path)
```

Ruff (0.15.10)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)